packages/cardano-graphql-services/src/Http/HttpServer.ts

```diff
@@ -1,27 +1,66 @@
+import { Logger, dummyLogger } from 'ts-log';
+import { ProviderError } from '@cardano-sdk/core';
 import { RunnableModule } from '../RunnableModule';
-import { dummyLogger } from 'ts-log';
 import { listenPromise, serverClosePromise } from '../util';
+import bodyParser from 'body-parser';
 import express from 'express';
+import expressPromBundle from 'express-prom-bundle';
 import http from 'http';
 import net from 'net';
+import promClient from 'prom-client';
 
 export type HttpServerConfig = {
-  router: express.Router;
-  name: string;
+  metrics?: {
+    enabled: boolean;
+    options?: expressPromBundle.Opts;
+  };
+  name?: string;
   listen: net.ListenOptions;
 };
 
+export interface HttpServerDependencies {
+  healthCheckCallback: () => Promise<{ ok: boolean }>;
+  router: express.Router;
+  logger?: Logger;
+}
+
 export abstract class HttpServer extends RunnableModule {
   public app: express.Application;
   public server: http.Server;
+  #dependencies: HttpServerDependencies;
 
-  protected constructor(public config: HttpServerConfig, logger = dummyLogger) {
-    super(config.name, logger);
+  protected constructor(public config: HttpServerConfig, { logger = dummyLogger, ...rest }: HttpServerDependencies) {
+    super(config.name || 'HttpServer', logger);
+    this.#dependencies = { logger, ...rest };
   }
 
   protected async initializeImpl(): Promise<void> {
+    this.#dependencies.router.use(bodyParser.json());
+    this.#dependencies.router.get('/health', async (req, res) => {
+      this.logger.debug('/health', { ip: req.ip });
+      let body: { ok: boolean } | Error['message'];
+      try {
+        body = await this.#dependencies.healthCheckCallback();
+      } catch (error) {
+        this.logger.error(error);
+        body = error instanceof ProviderError ? error.message : 'Unknown error';
+        res.statusCode = 500;
+      }
+      res.send(body);
+    });
     this.app = express();
-    this.app.use(this.config.router);
+    if (this.config.metrics?.enabled) {
+      this.app.use(
+        expressPromBundle({
+          includeMethod: true,
+          promRegistry: new promClient.Registry(),
+          ...this.config.metrics.options
+        })
+      );
+      this.logger.info(`Prometheus metrics configured at ${this.config.metrics.options?.metricsPath || '/metrics'}`);
```

Comment:
I'm wondering maybe the configured path is available on what `expressPromBundle` returns. Super minor, feel free to ignore this comment

Reply:
I did look for that before resorting to this clumsy solution, but will double-check.